code/_onclick/hud/screen_objects.dm

```diff
@@ -280,166 +336,195 @@
+	vis_contents += overlay_object
+
+/obj/effect/overlay/zone_sel
+	icon = 'icons/mob/screen/zone_sel.dmi'
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	alpha = 128
+	anchored = TRUE
+	layer = ABOVE_HUD_LAYER
+	plane = ABOVE_HUD_PLANE
 
+/obj/screen/zone_sel/proc/get_zone_at(icon_x, icon_y)
 	switch(icon_y)
 		if(1 to 3) //Feet
 			switch(icon_x)
 				if(10 to 15)
-					selecting = "r_foot"
+					return "r_foot"
 				if(17 to 22)
-					selecting = "l_foot"
-				else
-					return TRUE
+					return "l_foot"
 		if(4 to 9) //Legs
 			switch(icon_x)
 				if(10 to 15)
-					selecting = "r_leg"
+					return "r_leg"
 				if(17 to 22)
-					selecting = "l_leg"
-				else
-					return TRUE
+					return "l_leg"
 		if(10 to 13) //Hands and groin
 			switch(icon_x)
 				if(8 to 11)
-					selecting = "r_hand"
+					return "r_hand"
 				if(12 to 20)
-					selecting = "groin"
+					return "groin"
 				if(21 to 24)
-					selecting = "l_hand"
-				else
-					return TRUE
+					return "l_hand"
 		if(14 to 22) //Chest and arms to shoulders
 			switch(icon_x)
 				if(8 to 11)
-					selecting = "r_arm"
+					return "r_arm"
 				if(12 to 20)
-					selecting = "chest"
+					return "chest"
 				if(21 to 24)
-					selecting = "l_arm"
-				else
-					return TRUE
+					return "l_arm"
 		if(23 to 30) //Head, but we need to check for eye or mouth
 			if(icon_x in 12 to 20)
-				selecting = "head"
 				switch(icon_y)
 					if(23 to 24)
 						if(icon_x in 15 to 17)
-							selecting = "mouth"
+							return "mouth"
 					if(26) //Eyeline, eyes are on 15 and 17
 						if(icon_x in 14 to 18)
-							selecting = "eyes"
+							return "eyes"
 					if(25 to 27)
 						if(icon_x in 15 to 17)
-							selecting = "eyes"
+							return "eyes"
+				return "head"
+
+/obj/screen/zone_sel/proc/set_selected_zone(choice, mob/user)
+	if(isobserver(user))
+		return
 
-	if(old_selecting != selecting)
+	if(choice != selecting)
+		selecting = choice
 		update_icon(usr)
 	return TRUE
 
+/obj/screen/zone_sel/update_icon(mob/user)
+	cut_overlays()
+	add_overlay(mutable_appearance('icons/mob/screen/zone_sel.dmi', "[z_prefix][selecting]"))
+	user.zone_selected = selecting
 
 /obj/screen/zone_sel/alien
-	icon = 'icons/mob/screen1_alien.dmi'
-
+	icon = 'icons/mob/screen/alien.dmi'
+	z_prefix = "ay_"
 
 /obj/screen/zone_sel/robot
-	icon = 'icons/mob/screen1_robot.dmi'
-
+	icon = 'icons/mob/screen/cyborg.dmi'
 
 /obj/screen/healths
 	name = "health"
 	icon_state = "health0"
 	screen_loc = ui_health
-	icon = 'icons/mob/screen1_Midnight.dmi'
-
+	icon = 'icons/mob/screen/health.dmi'
 
 /obj/screen/healths/alien
-	icon = 'icons/mob/screen1_alien.dmi'
+	icon = 'icons/mob/screen/alien.dmi'
 	screen_loc = ui_alien_health
 
-
 /obj/screen/healths/robot
-	icon = 'icons/mob/screen1_robot.dmi'
+	icon = 'icons/mob/screen/cyborg.dmi'
 	screen_loc = ui_borg_health
 
+/obj/screen/healthdoll
+	name = "health doll"
+	icon = 'icons/mob/screen/health.dmi'
+	screen_loc = ui_healthdoll
+
+/*
+/obj/screen/healthdoll/Click()
```

Comment:
Is this for future implementation?

Reply:
Yep. Removing for now though, since it would be another hundred medical code lines at least and best done on another PR.